src/torchcodec/decoders/_video_decoder.py

```diff
@@ -183,13 +222,42 @@ def __init__(
             custom_frame_mappings=custom_frame_mappings_data,
         )
 
+        self._fallback_info = FallbackInfo()
+        self._has_decoded_frame = False
+
     def __len__(self) -> int:
         return self._num_frames
 
+    @property
+    def cpu_fallback(self) -> FallbackInfo:
+        # We can only determine whether fallback to CPU is happening when this
+        # property is accessed and requires that at least one frame has been decoded.
+        self._update_cpu_fallback()
+        return self._fallback_info
```

Comment:
2 nits:
- No need to define a separate `_update_cpu_fallback()` function, it can just be inlined here
- The convention (I *think* it's a convention??) is to use the same name for the `@property` and for the underlying cached object. That is, I think `self._fallback_info` should just be `self._cpu_fallback`. It makes it more obvious that it relates to the `@cpu_fallback` property.